app/src/main/java/com/duckduckgo/app/fire/fireproofwebsite/ui/FireproofWebsitesViewModel.kt

```diff
@@ -17,37 +17,45 @@
 package com.duckduckgo.app.fire.fireproofwebsite.ui
 
 import androidx.lifecycle.*
-import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteDao
+import androidx.lifecycle.Observer
 import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteEntity
+import com.duckduckgo.app.fire.fireproofwebsite.data.FireproofWebsiteRepository
 import com.duckduckgo.app.fire.fireproofwebsite.ui.FireproofWebsitesViewModel.Command.ConfirmDeleteFireproofWebsite
 import com.duckduckgo.app.global.DispatcherProvider
 import com.duckduckgo.app.global.SingleLiveEvent
+import com.duckduckgo.app.settings.db.SettingsDataStore
 import com.duckduckgo.app.statistics.pixels.Pixel
 import com.duckduckgo.app.statistics.pixels.Pixel.PixelName.FIREPROOF_WEBSITE_DELETED
 import kotlinx.coroutines.launch
+import java.util.*
 
 class FireproofWebsitesViewModel(
-    private val dao: FireproofWebsiteDao,
+    private val fireproofWebsiteRepository: FireproofWebsiteRepository,
     private val dispatcherProvider: DispatcherProvider,
-    private val pixel: Pixel
+    private val pixel: Pixel,
+    private val settingsDataStore: SettingsDataStore
 ) : ViewModel() {
 
     data class ViewState(
+        val loginDetectionEnabled: Boolean = false,
         val fireproofWebsitesEntities: List<FireproofWebsiteEntity> = emptyList()
     )
 
     sealed class Command {
         class ConfirmDeleteFireproofWebsite(val entity: FireproofWebsiteEntity) : Command()
     }
 
-    val viewState: MutableLiveData<ViewState> = MutableLiveData()
+    private val _viewState: MutableLiveData<ViewState> = MutableLiveData()
+    val viewState: LiveData<ViewState> = _viewState
```

Comment:
I wanted to introduce and see how it feels to add this guard against external collaborators having access to `MutableLiveData`. I introduced the change inside this class because it is small and "new".

I can rollback the change if you think it's not needed and it's adding noise to the class. No strong feelings to keep it.